### DISCUSSION TITLE
Global variables not consistent when using mockoon-cli that's deployed on GCP

### DISCUSSION THREAD

**Discussion by titibudala** [mockoon/mockoon#1610] [Help] [answered]
I'm trying to use the functionality that was introduced as part of this ticket: https://github.com/mockoon/mockoon/issues/1354

When using the Admin API to create a global variable and then again the Admin API to get that global variable, the response it's either the value that i've just set or "Global var not found". (using latest version of mockoon-cli that's deployed on GCP)

Curious if anyone ever had this kind of issue before. 👀 Locally it's working fine, but as soon as I deploy my changes, the "caching" issue occurs.

**Comment by 255kb** [comment]
One question first: when you say GCP, are you talking about a serverless-like solution (App Engine or Cloud Run)? Because in this case it's possible that the state of the mock (global variables and data buckets make the mocks stateful) is lost after the end of the HTTP connection.

**Reply by titibudala** [reply]
It's not a serverless solution if i'm not mistaken (client's infratructure) - it's using Google Kubernetes with docker images (thank you for respondig so quick)

**Reply by 255kb** [reply, marked as answer]
I just tested with a deployment on our cloud (https://mockoon.com/cloud/) and it's working correctly. We are hosting on GCP but with more traditional VMs on Compute Engine (also due to the stateful nature of the mocks we only deploy one instance of each). 
Cloud it be that the k8s cluster load balances multiple instances? 
I'm not a k8s specialist but I would recommend that the setup targets only one running instance (no up/down scaling or load balancing). This is what we are doing with the playground (https://playground.mockoon.com) using Cloud Run. Up scaling is kind of ok with sticky sessions, bu downscaling or load balancing without sticky session wouldn't work.

**Reply by titibudala** [reply]
You may be onto something, they are heavy on to load-balancing and microservices (will confirm with their devops team).
Thank you so much for the insights, will try to test/debug it and will be back with an answer in case it will help anyone in the future. Love ya!

**Reply by titibudala** [reply]
That was it! It was creating two instances by default and load-balacing between them. Now we create only one and it's working lovely! Thank you so much! 🍻

**Reply by 255kb** [reply]
great!
